### ISSUE TITLE
Chrome mainfest version

### ISSUE THREAD

**Issue by Xiaohuba** [jmerle/competitive-companion#321]
I downloaded this extension from [Release 2.38.2](https://github.com/jmerle/competitive-companion/releases/tag/2.38.2) and my chrome browser said:

```
Manifest version 2 is deprecated, and support will be removed in 2023. See https://developer.chrome.com/blog/mv2-transition/ for more details.
```

Since 2023 is coming, shall we update the manifest?

**Comment by jmerle** [comment]
I'm aware of the Manifest V3 update, but didn't know Chrome is starting to warn users like that. I'm intentionally not updating to Manifest V3 at the moment because Firefox doesn't support it yet and I don't want to maintain support for two manifest versions. Firefox [will support Manifest V3 starting with Firefox 109](https://blog.mozilla.org/addons/2022/11/17/manifest-v3-signing-available-november-21-on-firefox-nightly/) which is planned to be released January 17th, and Chrome is [probably not removing Manifest V2 support](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/) before that date, so I'm planning to update to Manifest V3 around the start of February (to give Firefox users time to update their browser).

**Comment by Xiaohuba** [comment]
Ok and thanks!

**Comment by jmerle** [comment]
The MV3 update is almost done in the https://github.com/jmerle/competitive-companion/tree/manifest-v3 branch, but is currently blocked on Firefox not yet supporting service workers as background pages, where-as Chrome forces background pages to be service workers. This can be fixed by having separate Chrome and Firefox distributions of Competitive Companion, but I want to hold off of that for as long as possible.

I'm still hoping Firefox will one day support service workers in background pages, [which a blog post from last year suggests as a possibility in the "near future"](https://blog.mozilla.org/addons/2022/10/31/begin-your-mv3-migration-by-implementing-new-features-today/) (emphasis mine):

> ### Will Chrome MV3 extensions work in Firefox MV2?
> The release of MV3 in Firefox is distinct from Chrome. Add-ons intended to work across different browsers will, in most cases, require some level of adjustment to be compatible in both Firefox and Chrome. **That said, we are committed to a high level of compatibility. We will be providing additional APIs and features in the near future.** If you’ve converted your Chrome extension to Google’s MV3, you may be able to consolidate some of those changes into your Firefox MV2 extension. Here are a few areas to investigate:
>
> - Service Workers are **not yet** available in Firefox; however many scripts may work interchangeably between Service Workers and Event Pages, depending on functionality. To get things working, you may need to remove service worker specific APIs. See [Service Worker Global Scope fo](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope)r more information.
> - DNR is not yet available in Firefox. Firefox retains WebRequest blocking in MV3, which can be used in place of DNR. When DNR is available, simple request modifications can be moved over to DNR.
> - The storage.session API is not yet available in Firefox. You can use other storage mechanisms in the meantime.

Furthermore, Chrome has paused it's MV2 phase-out plans, [with the latest update saying MV2 support will not be turned off without at least 6 months heads-up and won't happen until at least 2024](https://groups.google.com/u/0/a/chromium.org/g/chromium-extensions/c/zQ77HkGmK9E) (emphasis mine):
> We plan to continue reviewing feedback, making changes and improving documentation to ensure the transition from Manifest V2 to Manifest V3 is smooth and successful. We're still working on the timeline of the MV2 phase-out plan, so keep an eye out for it in the coming months. We will provide sufficient migration time for developers - at least 6 months of heads-up - before beginning any experiments to turn off MV2 in the browser next year.

Therefore, I'm going to wait a bit longer with moving Competitive Companion to MV3, in hopes of Firefox supporting service workers in background pages before Chrome decides to remove support for MV2.

**Comment by mightyhulk** [comment]
How can I use the competitive companion on interviewbit??

**Comment by jmerle** [comment]
> How can I use the competitive companion on interviewbit??

Please don't hijack issues with completely unrelated questions, but the answer is you can't. See #245 for more information.

**Comment by jmerle** [comment]
Implemented in 09088108a53f98bdb2f5b62c7b9995c8045fae37, the next release later this weekend will use Manifest V3.
